pkgs/misc/moonfire-nvr/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, rustPlatform
+, buildNpmPackage
+, fetchFromGitHub
+, pkg-config
+, ncurses
+, sqlite
+, testers
+, moonfire-nvr
+, breakpointHook
+}:
+
+let
+  pname = "moonfire-nvr";
+  version = "0.7.6";
+  src = fetchFromGitHub {
+    owner = "scottlamb";
+    repo = "moonfire-nvr";
+    rev = "v${version}";
+    hash = "sha256-hPgS4Y/dD6G8lqfsJz3aeeed6P+ngJpBOng88xUc55Q=";
+  };
+  ui = buildNpmPackage {
+    inherit version src;
+    pname = "${pname}-ui";
+    sourceRoot = "source/ui";
+    npmDepsHash = "sha256-IpZWgMo6Y3vRn9h495ifMB3tQxobLeTLC0xXS1vrKLA=";
+    installPhase = "
+      runHook preInstall
+
+      cp -r build $out
+
+      runHook postInstall
+    ";
+  };
+in rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  sourceRoot = "source/server";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "hashlink-0.8.1" = "sha256-h7DEapTVy0SSTaOV9rCkdH3em4A9+PS0k1QQh1+0P4c=";
+      "mp4-0.9.2" = "sha256-mJZJDzD8Ep9c+4QusyBtRoqAArHK9SLdFxG1AR7JydE=";
+    };
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    breakpointHook
+  ];
+
+  buildInputs = [
+    ncurses
+    sqlite
+  ];
+
+  postInstall = ''
+    mkdir -p $out/lib/ui
+    ln -s ${ui} $out/lib/ui
```

Comment:
sorry for the comment after this is merged, but how do you use the UI path that this creates?

For me, I have to specify `uiDir = "${pkgs.moonfire-nvr}/lib/ui/jhbw1r5q3vsp886vvg72wbgwaypp391x-moonfire-nvr-ui-0.7.7";` to get it to work.

This seems awkward and crucially like it is not a stable file path.

Is this a bug? But then I'm not sure how this was a bug as presumably this wasn't committed without being tried? :S

Reply:
Same problem. This seems wrong to me as well.

@gaelreyrol Can you clarify if this symlink was intended?

```
/nix/store/c0v8rfg69c042ck81vbmfhqgmlsbp2jd-moonfire-nvr-0.7.7/lib/ui/n9c8ijzh0ndgp6s5h88ix41wfhfxycpz-moonfire-nvr-ui-0.7.7 -> /nix/store/n9c8ijzh0ndgp6s5h88ix41wfhfxycpz-moonfire-nvr-ui-0.7.7/
```

Shouldn't it be

```
/nix/store/c0v8rfg69c042ck81vbmfhqgmlsbp2jd-moonfire-nvr-0.7.7/lib/ui -> /nix/store/n9c8ijzh0ndgp6s5h88ix41wfhfxycpz-moonfire-nvr-ui-0.7.7/
```

?

Reply:
I suspect it should be:

```sh
    mkdir -p $out/lib
    ln -s ${ui} $out/lib/ui
```